kong/utils.go

```diff
@@ -741,3 +741,119 @@ func FillPluginsDefaults(plugin *Plugin, schema Schema) error {
 func FillPluginsDefaultsWithOpts(plugin *Plugin, schema map[string]interface{}, opts FillRecordOptions) error {
 	return fillConfigRecordDefaultsAutoFields(plugin, schema, opts)
 }
+
+func deleteAndCollapseMap(config map[string]interface{}, path []string) {
+	key := path[0]
+	if len(path) == 1 {
+		delete(config, key)
+	} else {
+		if nested, ok := config[key].(map[string]interface{}); ok {
+			deleteAndCollapseMap(nested, path[1:])
+			if len(nested) == 0 {
+				delete(config, key)
+			}
+		}
+	}
+}
+
+func pathExistsInConfig(config map[string]interface{}, path []string) bool {
+	key := path[0]
+	if len(path) == 1 {
+		_, ok := config[key]
+		return ok
+	} else if nested, ok := config[key].(map[string]interface{}); ok {
+		return pathExistsInConfig(nested, path[1:])
+	}
+
+	return false
+}
+
+func clearUnmatchingDeprecationsHelper(
```

Comment:
Can we add comments over the functions to explain what they do and in which scenarios they may be useful?

Reply:
Done ✅